### PR TITLE
Migration guide

### PR DESCRIPTION
**Pull request by JordonPhillips**
This is based off of #1064, so only the last commit here needs to be looked at.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mtdowling** [comment]
I think setting the target branch to default-zero-value-docs would make this easier to review.

**Comment by JordonPhillips** [comment]
> I think setting the target branch to default-zero-value-docs would make this easier to review.

That branch is on my fork, so that's not possible. Since there is only a singular unique commit, you can just click on it to review that commit. I could also just rebase on idl-2.0 but then there'd be a ton of broken links.

File: docs/source/1.0/guides/migrationg-idl-1-to-2.rst [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+
+Replace Primitive prelude shape targets
+=======================================
+
+The primitive shapes have been removed from the prelude, and so any member
+targeting one of them must update to target its equivalent non-primitive
+shape.
+
+.. list-table:
```
Comment (mtdowling) [inline docs/source/1.0/guides/migrationg-idl-1-to-2.rst:83]: Whe updating structure members that target these, they'll need to add the `@default` trait.

File: docs/source/1.0/guides/migrationg-idl-1-to-2.rst [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+a single combined model. Therefore when migrating, it may be wise to migrate
+one file at a time.
+
+
+Remove the box trait
+====================
+
+The ``box`` trait is removed in 2.0, so it must be removed from any shapes or
+members that use it. Smithy structure members are considered boxed by default,
+which can be changed using the :ref:`required-trait` or :ref:`default-trait`.
```
Comment (mtdowling) [inline docs/source/1.0/guides/migrationg-idl-1-to-2.rst:69]: Maybe add this?
> In effect, this means you can indiscriminately remove the `@box` trait from your models.

File: docs/source/1.0/guides/migrationg-idl-1-to-2.rst [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+Similarly, :ref:`mixins` can be useful if you have a shared set of traits
+that otherwise have to be copied and pasted.
+
+.. seealso::
+
+    the :ref:`mixins section <mixins>` of the spec for more details on how they
+    work.
+
+
+Remove unsightly commas
```
Comment (mtdowling) [inline docs/source/1.0/guides/migrationg-idl-1-to-2.rst:252]: 😆

File: docs/source/1.0/guides/migrationg-idl-1-to-2.rst [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+
+    operation GetUser {
+        input: GetUserInput,
+        output: GetUserOutput,
+        errors: [
+            NotFoundError,
+            AccessDeniedError,
+        ],
+    }
+
```
Comment (mtdowling) [inline docs/source/1.0/guides/migrationg-idl-1-to-2.rst:272]: Nit: can just remove this line and the next in both this example and the next example

File: docs/source/1.0/guides/migrationg-idl-1-to-2.rst [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+    the :ref:`inline input / output <idl-inline-input-output>` section of the
+    spec for more details.
+
+
+Abstract shared shape configuration with mixins
+-----------------------------------------------
+
+Models often have several shapes that refer to the same sets of members, or
+which share a set of trait configurations. For example, resource instance
+operations all require that the resource's identifiers be present. With
```
Comment (mtdowling) [inline docs/source/1.0/guides/migrationg-idl-1-to-2.rst:173]: *in input*

File: docs/source/1.0/guides/migrationg-idl-1-to-2.rst [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+            email: String
+            id: String
+            username: String
+        },
+        read: GetUser
+    }
+
+    @mixin
+    structure UserIdentifiers {
+        email: String
```
Comment (mtdowling) [inline docs/source/1.0/guides/migrationg-idl-1-to-2.rst:231]: These members all need to be required... Might be nice to just trim down the example to one identifier too.

**Review by mtdowling — CHANGES_REQUESTED** [review]

File: docs/source/1.0/guides/migrationg-idl-1-to-2.rst (hunk 6 above)
Comment (JordonPhillips) [reply docs/source/1.0/guides/migrationg-idl-1-to-2.rst:231]: How about two? I want there to be more than one to emphasize how annoying it is to copy and paste.

**Review by mtdowling — APPROVED** [review]

**Review by mtdowling — APPROVED** [review]

**Pull request merged**